packages/react-storage/src/components/StorageBrowser/views/LocationActionView/hooks/useDeleteActionView.tsx

```diff
@@ -0,0 +1,219 @@
+import { UseActionView } from './types';
+import {
+  getActionIconVariant,
+  getActionViewTaskStatuses,
+  getFilenameWithoutPrefix,
+  getFileTypeDisplayValue,
+} from './utils';
+
+interface UseDeleteActionView extends UseActionView {}
+
+const LOCATION_ACTION_VIEW_HEADERS: DataTableProps['headers'] = [
+  { key: 'key', type: 'sort', content: { label: 'Name' } },
+  { key: 'folder', type: 'text', content: { text: 'Folder' } },
+  { key: 'type', type: 'text', content: { text: 'Type' } },
+  { key: 'size', type: 'text', content: { text: 'Size' } },
+  { key: 'status', type: 'sort', content: { label: 'Status' } },
+  { key: 'action', type: 'text', content: { text: '' } },
+];
+
+export const getDeleteActionViewTableData = ({
+  tasks,
+  taskCounts,
+  path,
+}: {
+  tasks: Task[];
+  taskCounts: TaskCounts;
+  path: string;
+}): DataTableProps => {
+  const { hasStarted } = getActionViewTaskStatuses(taskCounts);
+
+  const rows: DataTableProps['rows'] = tasks.map((item) => {
+    const row: WithKey<DataTableRow> = {
+      key: item.key,
+      content: LOCATION_ACTION_VIEW_HEADERS.map(({ key: columnKey }) => {
+        const key = `${columnKey}-${item.key}`;
+        switch (columnKey) {
+          case 'key': {
+            return {
+              key,
+              type: 'text',
+              content: {
+                icon: getActionIconVariant(item.status),
+                text: getFilenameWithoutPrefix(item.key),
+              },
+            };
+          }
+          case 'folder': {
+            return {
+              key,
+              type: 'text',
+              content: {
+                text: path,
+              },
+            };
+          }
+          case 'type': {
+            return {
+              key,
+              type: 'text',
+              content: {
+                text: getFileTypeDisplayValue(item.key),
+              },
+            };
+          }
+          case 'size':
+            return {
+              key,
+              type: 'text',
+              content: {
+                text: humanFileSize(
+                  parseInt((item as Task & { size: string }).size),
+                  true
+                ),
+              },
+            };
+          case 'status':
+            return {
+              key,
+              type: 'text',
+              content: {
+                text: STATUS_DISPLAY_VALUES[item.status],
+              },
+            };
+          case 'action':
+            // don't allow removing a single task
+            if (taskCounts.TOTAL > 1) {
+              return hasStarted
+                ? {
+                    key,
+                    type: 'button',
+                    content: {
+                      icon: 'cancel',
+                      ariaLabel: `Cancel item: ${item.key}`,
+                      onClick: () => item.cancel?.(),
+                      disabled: item.status !== 'QUEUED',
+                    },
```

Comment:
Don't think component props should be returned from the hook?

Reply:
moved tabledata down to be built in the view control